internal/cni/cni.go

```diff
@@ -344,12 +345,14 @@ func CmdDel(args *skel.CmdArgs) error {
 		}
 	}
 
-	logging.Infof("cmdDel(): removing BPF program from device")
-	if err := bpfHandler.Cleanbpf(cfg.Device); err != nil {
-		err = fmt.Errorf("cmdDel(): error removing BPF program from device: %w", err)
-		logging.Errorf(err.Error())
+	if cfg.UnloadBpf {
+		logging.Infof("cmdDel(): removing BPF program from device")
+		if err := bpfHandler.Cleanbpf(cfg.Device); err != nil {
+			err = fmt.Errorf("cmdDel(): error removing BPF program from device: %w", err)
+			logging.Errorf(err.Error())
 
-		return err
+			return err
+		}
```

Comment:
I'm fine with making this optional, what I don't like is it's changing the default behavior.

Right now, default behavior is to clean up the BPF.
However, the default value of the Boolean is false, meaning if the user doesn't add this new config to their NAD, then their CNI will no longer clean up - default behavior has changed.

Two ways around this:

1. Above around line 100. `n := &NetConfig{}` could become `n := &NetConfig{UnloadBpf:true}` giving the config the default value of true. I think that'd work, would need to test the behavior.
2. We could changing the "wording" of the config setting here. Something like `SkipUnloadBpf` so it makes sense for false to remain the default value. Update line 348 here then to be something like `if cfg.SkipUnloadBpf != false`.

I'm leaning towards option 2, but either way, it means the user only needs to update their config if they have that edge usecase where they want "raw" subfunctions.

You could then remove all changes in `test/e2e/nad.yaml` and `.github/e2e/nad.yaml`